code/mission/import/xwinglib.cpp

```diff
@@ -582,5 +599,161 @@ bool XWingMission::load(XWingMission *m, const char *data)
 		m->flightgroups.push_back(*nfg);
 	}
 
+	for (int n = 0; n < h->number_of_objects; n++) {
+		xwi_objectgroup *oj =
+			(xwi_objectgroup*)(data + sizeof(xwi_header) + sizeof(xwi_flightgroup) + sizeof(xwi_objectgroup) * n);
+		XWMObject noj_buf;
+		XWMObject *noj = &noj_buf;
+
+		noj->designation = oj->designation;
+		noj->cargo = oj->cargo;
+		noj->specialCargo = oj->special_cargo;
+		noj->specialObjectNumber = oj->special_object_number;
+
+		switch (oj->object_type) {
```

Comment:
Look at the `CraftType` list in the file format.  Mine1 is hexadecimal 12 (decimal 18) and the other object types likewise need their indexes fixed.